ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js

```diff
@@ -141,8 +176,16 @@ export default class ConfirmApproveContent extends Component {
       isMultiLayerFeeNetwork,
       supportsEIP1559,
     } = this.props;
-    if (!isMultiLayerFeeNetwork && supportsEIP1559) {
-      return <GasDetailsItem />;
+    const { userAcknowledgedGasMissing } = this.state;
+    const hasSimulationError = Boolean(txData.simulationFails);
+    const renderSimulationFailureWarning =
+      hasSimulationError && !userAcknowledgedGasMissing;
+    if (
+      !isMultiLayerFeeNetwork &&
+      supportsEIP1559 &&
+      !renderSimulationFailureWarning
+    ) {
+      return <GasDetailsItem userAcknowledgedGasMissing />;
```

Comment:
A re-usable hook to get this value `renderSimulationFailureWarning` might be useful. I see this logic repeated a few places.

Reply:
@jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17437/commits/cabe23bdddf6470922c9215baf57b396a4a8e953).